### DISCUSSION TITLE
ROS2 Support?

### DISCUSSION THREAD

**Discussion by smnogar** [foxglove/community#43] [Q&A]
Just stumbled upon this. Its really great! I love the drag and drop GUI and the ability to scroll around the timeline.

I'm in the process of transitioning to ROS2, any plans on supporting it?

**Comment by jtbandes** [comment]
Hi! Thanks for trying out the product and getting in touch!

We are planning to start adding support for ROS 2 bags and direct ROS 2 connections soon. This and more on our roadmap here: https://discourse.ros.org/t/foxglove-studio-visualization-and-debugging-for-ros/19627/6

I'd be interested if you could share more about how you're using ROS 2 - are you primarily working with recorded bag files, or live robots? Are you using the rosbridge WebSocket server at all? This can help us prioritize what we work on first :)

**Reply by smnogar** [reply]
Hi! We are in the process of transitioning our UAS to use ROS2. From my perspective we would probably use this as a way to view bag files, rather than live data. Rviz is probably a little better suited for us for realtime viewing. That said, this is a really nice tool, so maybe we would consider using it live as well. I've never used the rosbridge web socket, but if that's the preferred interface to ros2, we could run it.

A small feature request, can the map view be optionally a satellite view? Not all robots travel on roads! :)

**Reply by jtbandes** [reply]
You can now follow #715 (native connection), #716 (websocket connection), #717 (bag files) for tracking the various ROS 2 compatibility features!

As for satellite view, I just created a new issue to track that: https://github.com/foxglove/studio/issues/1182

**Reply by jtbandes** [reply]
Update: we've just released ROS 2 bag support with Studio v0.13.0: https://foxglove.dev/blog/announcing-ros2-bag-support
And ROS 2 live connections in v0.18: https://foxglove.dev/blog/announcing-live-ros2-support
